Modules/posixmodule.c

```diff
@@ -10035,6 +10040,175 @@ os_times_impl(PyObject *module)
 #endif /* HAVE_TIMES */
 
 
+#if defined(HAVE_TIMERFD_API)
+#define ONE_SECOND_IN_NS (1000 * 1000 * 1000)
+#define EXTRACT_NSEC(value)  (long)( ( (double)(value) - (time_t)(value) ) * 1e9)
+
+static PyObject *
+build_itimerspec(const struct itimerspec* curr_value)
+{
+    return PyTuple_Pack(
+        2,
+        PyFloat_FromDouble((double)(curr_value->it_interval.tv_sec) + (double)curr_value->it_interval.tv_nsec * 1e-9),
+        PyFloat_FromDouble((double)(curr_value->it_value.tv_sec) + (double)curr_value->it_value.tv_nsec * 1e-9)
+    );
```

Comment:
You'll need to add error handling and dereference temporary objects:

```suggestion
    double _interval = (double)(curr_value->it_interval.tv_sec) +
                       (double)curr_value->it_interval.tv_nsec * 1e-9;
    PyObject *interval = PyFloat_FromDouble(_inverval);
    if (interval == NULL) {
        return NULL;
    }
    double _value = (double)(curr_value->it_value.tv_sec) +
                    (double)curr_value->it_value.tv_nsec * 1e-9;
    PyObject *value = PyFloat_FromDouble(_value);
    if (value == NULL) {
        Py_DECREF(interval);
        return NULL;
    }
    PyObject *tuple = PyTuple_Pack(2, interval, value);
    Py_DECREF(interval);
    Py_DECREF(value);
    return tuple;
```

Ditto for similar functions.